### PR TITLE
Prevent admin to overwrite Stock Data when order placed while he was …

### PR DESCRIPTION
**Pull request by seruymt**
https://github.com/magento-engcom/msi/issues/538

*Need to fix case for Magento CE without MSI

### CODE REVIEW

File: app/code/Magento/CatalogInventory/view/adminhtml/web/js/components/stock-status.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+                }
+
+                return false;
+            });
+
+            return this.normalizeData(value);
+        },
+
+        /** @inheritdoc */
+        validate: function () {
```
Comment (omiroshnichenko) [inline app/code/Magento/CatalogInventory/view/adminhtml/web/js/components/stock-status.js]: In original fix logic was implemented in `setDifferedFromDefault` method. Why you move it in `validate`?

File: app/code/Magento/InventoryCatalog/Observer/ProcessSourceItemsObserver.php [hunk 2]
```diff
@@ -58,14 +93,57 @@ public function execute(EventObserver $observer)
...
         /** @var Save $controller */
         $controller = $observer->getEvent()->getController();
+        $productData = $controller->getRequest()->getParam('product', []);
+        $singleSourceData = $productData['quantity_and_stock_status'] ?? [];
 
-        $sources = $controller->getRequest()->getParam('sources', []);
-        $assignedSources = isset($sources['assigned_sources']) && is_array($sources['assigned_sources'])
-            ? $sources['assigned_sources'] : [];
+        if (!$this->isSingleSourceMode->execute()) {
+            $sources = $controller->getRequest()->getParam('sources', []);
```
Comment (maghamed) [inline app/code/Magento/InventoryCatalog/Observer/ProcessSourceItemsObserver.php:100]: btw have you tested the case when you are in Multi Stock Mode, but did not fill any Source Item data? So, `'sources'` will be empty.
Or is it always has nested `'assigned_sources'` ?

Comment (seruymt) [reply app/code/Magento/InventoryCatalog/Observer/ProcessSourceItemsObserver.php:100]: In this case "source" is not in the POST array

**Comment by seruymt** [comment]
1. Replace quantity_and_stock validation scripts from CatalogInventory to MSI InventoryCatalog (depends on #571) - @omiroshnichenko also, please check it
2. Add fix for empty qty in POST request

**Review by maghamed — APPROVED** [review]

**Pull request merged**